api-metastore/src/main/java/org/zalando/nakadi/validation/schema/SchemaTypeChangeConstraint.java

```diff
@@ -0,0 +1,18 @@
+package org.zalando.nakadi.validation.schema;
+
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.EventTypeBase;
+
+import java.util.Optional;
+
+public class SchemaTypeChangeConstraint implements SchemaEvolutionConstraint{
+
+    @Override
+    public Optional<SchemaEvolutionIncompatibility> validate(final EventType original, final EventTypeBase eventType) {
+        if(!original.getSchema().getType().equals(eventType.getSchema().getType())){
```

Comment:
how do we plan to migrate from json to avro?

Reply:
I put this in place as I am not sure what things will break on allowing such a change, this is a temporary measure until we figure out things.